data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/worker/OpenSearchIndexPartitionCreationSupplier.java

```diff
@@ -70,13 +78,37 @@ private List<PartitionIdentifier> applyForOpenSearchClient(final Map<String, Obj
         }
 
         return indicesResponse.valueBody().stream()
-                .filter(this::shouldIndexBeProcessed)
+                .filter(osIndicesRecord -> shouldIndexBeProcessed(osIndicesRecord, null))
                 .map(indexRecord -> PartitionIdentifier.builder().withPartitionKey(indexRecord.index()).build())
                 .collect(Collectors.toList());
     }
 
-    private boolean shouldIndexBeProcessed(final IndicesRecord indicesRecord) {
-        if (Objects.isNull(indicesRecord.index())) {
+    private List<PartitionIdentifier> applyForElasticSearchClient(final Map<String, Object> globalStateMap) {
+        co.elastic.clients.elasticsearch.cat.IndicesResponse indicesResponse;
+        try {
+            indicesResponse = elasticsearchClient.cat().indices();
+        } catch (IOException | ElasticsearchException e) {
+            LOG.error("There was an exception when calling /_cat/indices to create new index partitions", e);
+            return Collections.emptyList();
+        }
+
+        return indicesResponse.valueBody().stream()
+                .filter(esIndicesRecord -> shouldIndexBeProcessed(null, esIndicesRecord))
+                .map(indexRecord -> PartitionIdentifier.builder().withPartitionKey(indexRecord.index()).build())
+                .collect(Collectors.toList());
+    }
+
+    private boolean shouldIndexBeProcessed(final IndicesRecord openSearchIndicesRecord, final co.elastic.clients.elasticsearch.cat.indices.IndicesRecord elasticSearchIndicesRecord) {
```

Comment:
We can just take in the index name here to simplify the logic.

The calls above to this method would then be `filter(record -> shouldIndexBeProcessed(record.index()))`